### PR TITLE
Upgrades storybook packages 

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
Upgrades Storybook packages in root workspace, `@lg-tools/storybook` and `@lg-tools/storybook-decorators` to latest

Additionally, removes unused storybook-related dependencies in various packages

### CODE REVIEW

**Review by shaneeza — APPROVED** [review]

File: .changeset/calm-shirts-smell.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@lg-tools/storybook-decorators': patch
+'@lg-tools/storybook': patch
+---
+
+Upgragdes Storybook packages to v7.6.x
```
Comment (stephl3) [inline .changeset/calm-shirts-smell.md:6]: nit: Upgrades*

do we usually specify the patch version? 7.6.17 vs 7.6.x?

File: .changeset/brave-rings-give.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@leafygreen-ui/date-picker': patch
+'@leafygreen-ui/toast': patch
+---
+
+Removes unused storybook dev dependencies (`@storybook/types` and `storybook-mock-date-decorator`)
```
Comment (stephl3) [inline .changeset/brave-rings-give.md:6]: thanks for cleaning those out 🙏

**Review by stephl3 — APPROVED** [review]

File: .changeset/calm-shirts-smell.md (hunk 1 above)
Comment (TheSonOfThomp) [reply .changeset/calm-shirts-smell.md:6]: Depends whether the patch version is relevant. I omitted it here since it's the 7.6 that's most important

**Review by stephl3 — APPROVED** [review]

File: .changeset/calm-shirts-smell.md (hunk 1 above)
Comment (TheSonOfThomp) [reply .changeset/calm-shirts-smell.md]: ... but we are using static versions, so I updated this message

**Pull request merged**